File: docs/code-examples/arrow2d_simple.cpp
// Log a batch of 2D arrows.

#include <rerun.hpp>

int main() {
    const auto rec = rerun::RecordingStream("rerun_example_arrow2d");
    rec.spawn().exit_on_failure();

    rec.log(
        "arrows",
        rerun::Arrows2D::from_vectors({{1.0f, 0.0f}, {0.0f, -1.0f}, {-0.7f, 0.7f}})
            .with_radii(0.025f)
            .with_origins({{0.0f, 0.0f}})
            .with_colors({{255, 0, 0}, {0, 255, 0}, {127, 0, 255}})
            .with_labels({"right", "up", "left-down"})
    );
}

File: docs/code-examples/arrow2d_simple.rs
//! Log a batch of 2D arrows.

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let rec = rerun::RecordingStreamBuilder::new("rerun_example_arrow2d").spawn()?;

    rec.log(
        "arrows",
        &rerun::Arrows2D::from_vectors([[1.0, 0.0], [0.0, -1.0], [-0.7, 0.7]])
            .with_radii([0.025])
            .with_origins([rerun::Position2D::ZERO])
            .with_colors([[255, 0, 0], [0, 255, 0], [127, 0, 255]])
            .with_labels(["right", "up", "left-down"]),
    )?;

    Ok(())
}
